integration/spring-boot/autoconfigure/src/main/java/org/apache/cxf/spring/boot/autoconfigure/micrometer/MicrometerMetricsAutoConfiguration.java

```diff
@@ -0,0 +1,144 @@
+@ConditionalOnBean({MeterRegistry.class})
+@EnableConfigurationProperties(CxfProperties.class)
+public class MicrometerMetricsAutoConfiguration {
+
+    private final CxfProperties properties;
+
+    public MicrometerMetricsAutoConfiguration(CxfProperties properties) {
+        this.properties = properties;
+    }
+
+    @Bean
+    public TimedAnnotationProvider timedAnnotationProvider() {
+        return new SpringBasedTimedAnnotationProvider();
+    }
+
+    @Bean
+    public JaxwsTags jaxwsTags() {
+        return new JaxwsTags();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(ExceptionClassProvider.class)
+    public ExceptionClassProvider exceptionClassProvider() {
+        return new DefaultExceptionClassProvider();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(JaxwsFaultCodeProvider.class)
+    public JaxwsFaultCodeProvider jaxwsFaultCodeProvider() {
+        return new JaxwsFaultCodeProvider();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(JaxwsFaultCodeTagsCustomizer.class)
+    public JaxwsFaultCodeTagsCustomizer jaxwsFaultCodeTagsCustomizer(JaxwsTags jaxwsTags,
+                                                                     JaxwsFaultCodeProvider jaxwsFaultCodeProvider) {
+        return new JaxwsFaultCodeTagsCustomizer(jaxwsTags, jaxwsFaultCodeProvider);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(JaxwsOperationTagsCustomizer.class)
+    public JaxwsOperationTagsCustomizer jaxwsOperationTagsCustomizer(JaxwsTags jaxwsTags) {
+        return new JaxwsOperationTagsCustomizer(jaxwsTags);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(StandardTags.class)
+    public StandardTags standardTags() {
+        return new StandardTags();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(TagsProvider.class)
+    public TagsProvider tagsProvider(ExceptionClassProvider exceptionClassProvider, StandardTags standardTags) {
+        return new StandardTagsProvider(exceptionClassProvider, standardTags);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(MetricsProvider.class)
+    public MetricsProvider metricsProvider(TagsProvider tagsProvider,
+                                           List<TagsCustomizer> tagsCustomizers,
+                                           TimedAnnotationProvider timedAnnotationProvider,
+                                           MeterRegistry registry) {
+        MicrometerMetricsProperties micrometerMetricsProperties = new MicrometerMetricsProperties();
+
+        Server server = this.properties.getMetrics().getServer();
+        micrometerMetricsProperties.setAutoTimeRequests(server.isAutoTimeRequests());
+        micrometerMetricsProperties.setRequestsMetricName(server.getRequestsMetricName());
+
+        return new MicrometerMetricsProvider(registry, tagsProvider, tagsCustomizers, timedAnnotationProvider,
+                micrometerMetricsProperties);
+    }
+
+    @Bean
+    @Order(0)
+    public MeterFilter metricsSoapServerUriTagFilter() {
```

Comment:
I think we could strip `soap` from the bean name, may be name it as `metricsMaxAllowedServerUriTagsFilter`

Reply:
I rename it to `cxfMetricsMaxAllowedServerUriTagsFilter`